frontend/components/questions/question-form.tsx

```diff
@@ -0,0 +1,116 @@
+"use client";
+
+import { Question } from "@/lib/schemas/question-schema";
+import { useEffect, useState } from "react";
+import {
+  Card,
+  CardContent,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "../ui/card";
+import { Input } from "../ui/input";
+import { Textarea } from "../ui/textarea";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "../ui/select";
+import { Button } from "../ui/button";
+import { Label } from "../ui/label";
+
+interface QuestionFormProps {
+  data: Question | undefined;
+  isAdmin: boolean | undefined;
+  handleSubmit?: (question: Question) => void;
+}
+
+const QuestionForm: React.FC<QuestionFormProps> = ({ ...props }) => {
+  const [question, setQuestion] = useState<Question>();
+
+  useEffect(() => {
+    setQuestion(props.data);
+  }, [props.data]);
+
+  return (
+    <form>
+      <Card>
+        <CardHeader>
+          <CardTitle className="m-4">
+            <Label>Title</Label>
+            <Input
+              value={question?.title}
+              className="mt-2"
+              onChange={(e) =>
+                question && setQuestion({ ...question, title: e.target.value })
+              }
+              disabled={!props.isAdmin}
+            />
+          </CardTitle>
+        </CardHeader>
+        <CardContent>
+          <div className="m-4">
+            <Label>Category</Label>
+            <Input
+              value={question?.category}
+              className="mt-2"
+              onChange={(e) =>
+                question &&
+                setQuestion({ ...question, category: e.target.value })
+              }
+              disabled={!props.isAdmin}
+            />
+          </div>
+          <div className="m-4">
+            <Label>Complexity</Label>
+            <div className="mt-2">
+              <Select
+                value={question?.complexity}
+                onValueChange={(e) =>
+                  question && setQuestion({ ...question, complexity: e })
+                }
+                disabled={!props.isAdmin}
+              >
+                <SelectTrigger id="complexity">
+                  <SelectValue placeholder="Select complexity" />
+                </SelectTrigger>
+                <SelectContent>
+                  <SelectItem value="easy">Easy</SelectItem>
+                  <SelectItem value="medium">Medium</SelectItem>
+                  <SelectItem value="hard">Hard</SelectItem>
+                </SelectContent>
+              </Select>
+            </div>
+          </div>
+          <div className="m-4">
+            <Label>Description</Label>
+            <Textarea
```

Comment:
The text area for the question description should be made longer too. Tested with a sample long question description and user needs to scroll down a lot to view the whole question, which might not be user-experience friendly

Reply:
Will use an auto resizable `textarea` component from [shadcn extensions](https://shadcnui-expansions.typeart.cc/docs/autosize-textarea)